### DISCUSSION TITLE
MasterDetail EnumKey

### DISCUSSION THREAD

**Discussion by havvanur160401046** [serenity-is/Serenity#5587] [Help]
![image](https://user-images.githubusercontent.com/24252615/111472416-fac50180-873a-11eb-9023-155c2ba79d04.png)
The Stok Türü type value is enum type and the enum value comes after saving the invoice. How can I get it when every detail is added?
![image](https://user-images.githubusercontent.com/24252615/111472513-14fedf80-873b-11eb-9ca0-bead891324ca.png)
![image](https://user-images.githubusercontent.com/24252615/111471999-973ad400-873a-11eb-8546-71ea6a47ba7a.png)
![image](https://user-images.githubusercontent.com/24252615/111472028-9efa7880-873a-11eb-8eda-be21aac01be3.png)

I guess I need to do it within the valideentity function, but how can I

**Comment by erehtolleh** [comment]
It seems like you already answered your own post.
Have you tried below?  I believe it should go in the validateentity in xyzChildEditor.ts file.  If you don' t have a xyzChildEditor.ts file, you will have to create one.  Sergen doesn't create it by default.  See #5390 

`row.StokStokName = YummurtaKind[row.StokId]`
